src/Button.php

```diff
@@ -4,95 +4,425 @@
     {
         $new = clone $this;
-        $new->tagName = $value;
+        $new->tag = match ($value) {
+            ButtonType::LINK => A::tag(),
+            ButtonType::RESET => Input::resetButton('Reset'),
+            ButtonType::SUBMIT => Input::submitButton('Submit'),
+        };
 
         return $new;
     }
+
+    /**
+     * Sets the URL of the link button.
+     *
+     * @param string|null $value The URL of the link button.
+     *
+     * @return self A new instance with the specified URL.
+     */
+    public function url(string|null $value): self
+    {
+        $new = clone $this;
+        $new->attributes['href'] = $value;
+
+        return $new;
+    }
+
+    /**
+     * Set the button variant. The following options are allowed:
+     *
+     * - `ButtonVariant::PRIMARY`: Primary button.
+     * - `ButtonVariant::SECONDARY`: Secondary button.
+     * - `ButtonVariant::SUCCESS`: Success button.
+     * - `ButtonVariant::DANGER`: Danger button.
+     * - `ButtonVariant::WARNING`: Warning button.
+     * - `ButtonVariant::INFO`: Info button.
+     * - `ButtonVariant::LIGHT`: Light button.
+     * - `ButtonVariant::DARK`: Dark button.
+     *
+     * @param ButtonVariant $value The button variant.
+     *
+     * @return self A new instance with the specified button variant.
+     */
+    public function variant(ButtonVariant $value): self
+    {
+        $new = clone $this;
+        $new->buttonVariant = $value;
+
+        return $new;
+    }
+
+    /**
+     * Run the button widget.
+     *
+     * @return string The HTML representation of the element.
+     */
+    public function render(): string
+    {
+        $attributes = $this->attributes;
+        $classes = $attributes['class'] ?? null;
+        $tag = $this->tag ?? ButtonTag::tag()->button('');
+
+        $id = match ($this->id) {
+            true => $attributes['id'] ?? Html::generateId(self::NAME . '-'),
+            '', false => null,
+            default => $this->id,
+        };
+
+        unset($attributes['class'], $attributes['id']);
+
+        Html::addCssClass($attributes, [self::NAME, $this->buttonVariant->value, $classes, ...$this->cssClass]);
+
+        $attributes = $this->setAttributes($attributes);
+
+        if ($tag instanceof Input) {
+            if ($this->label !== '') {
+                $tag = $tag->value($this->label);
+            }
+
+            return $tag->addAttributes($attributes)->id($id)->render();
+        }
+
+        return $tag->addAttributes($attributes)->addContent($this->label)->id($id)->encode(false)->render();
+    }
+
+    /**
+     * Sets the attributes for the button.
+     *
+     * @param array $attributes The attributes to set.
+     *
+     * @return array The updated attributes.
+     */
+    private function setAttributes(array $attributes): array
+    {
+        if ($this->active) {
+            $attributes['aria-pressed'] = 'true';
+            $attributes['data-bs-toggle'] = 'button';
```

Comment:
It's should be optional and disabled by default.

Reply:
Attributes are only configured in the case that is active, as are the documents.

Reply:
"active" is state of button and can used without toggle.

1) Case 1. Add class "active" only.

2) Case 2. Add class "active" and `data-bs-toggle="button" aria-pressed="true"`

3) Case 3. Add `data-bs-toggle="button"` only.

So we need two methods: "active()" (add/remove "active" class), "toggleActive()" (enable/disable behavior of "active" toggle).

May be "toggleActive" is not good name...

Reply:
Case 1: According to the documentation when it is active or disabled you should always add data-bs-toggle.
Case 3: Fixed.